checker/src/features/constant_functions.rs

```diff
@@ -306,6 +312,153 @@ pub(crate) fn call_constant_function(
 			crate::utilities::notify!("TODO json:stringify");
 			Err(ConstantFunctionError::BadCall)
 		}
+		"regexp:constructor" => {
+			let pattern = types
+				.get_type_by_id(arguments.first().unwrap().non_spread_type().expect("pattern"));
+			let flags =
+				arguments.get(1).map(|a| types.get_type_by_id(a.non_spread_type().expect("flags")));
+
+			let Type::Constant(Constant::String(pattern)) = pattern else {
+				return Err(ConstantFunctionError::BadCall);
+			};
+			let flags = match flags {
+				Some(flags) => {
+					let Type::Constant(Constant::String(flags)) = flags else {
+						return Err(ConstantFunctionError::BadCall);
+					};
+
+					Some(flags.clone())
+				}
+				None => None,
+			};
+
+			let regex =
+				types.new_regex(&pattern.clone(), &flags, &call_site.clone().without_source());
+
+			Ok(ConstantOutput::Value(regex))
+		}
+		"regexp:exec" => {
+			let this = this_argument.get_passed().map(|t| types.get_type_by_id(t));
+
+			if let Some(Type::SpecialObject(SpecialObjects::RegularExpression(regexp))) = this {
+				let pattern_type_id =
+					arguments.first().unwrap().non_spread_type().expect("pattern");
+				let pattern_type = types.get_type_by_id(pattern_type_id);
+				let pattern = {
+					let Type::Constant(Constant::String(pattern)) = pattern_type else {
```

Comment:
This is great, if the `exec` is run with a constant string argument then we get a constant result.

I assume [this example](https://www.typescriptlang.org/play/?ssl=7&ssc=2&pln=1&pc=1#code/JYMwBAFAdgpg7mASjA5gUQB4AcICIAWwuAlAHQwYwDGehJxYA3gFDMC+roksCy62tImQrU8AIwCGAE3pNWbIA) gets always true or false warnings under this branch (where TSC doesn't)

I wonder whether this can be extended for general (ie via Type::RootPolyType or Type::Constructor) strings not through the constant result but so that `.groups` has named members. So `new RegExp("Hi (?<name>.*)").exec(event.target.value)` would have groups as { name: string }. I think TS added this, but I can't find it?

Reply:
Oh wait, I think this is being collected below? https://github.com/kaleidawave/ezno/pull/184/files#diff-be53d15efbd9ffdb15e107f75ef4867a569eb1ba61ecbd6cc1c468d3f49e8193R53

I will have a think about how this could work

Reply:
The TS Playground example works:

![image](https://github.com/user-attachments/assets/524c0792-0602-4f27-ac56-4dd2b8ba799d)